test/framework/src/test/java/org/elasticsearch/test/AbstractXContentTestCaseTests.java

```diff
@@ -0,0 +1,52 @@
+package org.elasticsearch.test;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class AbstractXContentTestCaseTests extends ESTestCase {
+
+    private static final int NUMBER_OF_TEST_RUNS = 10;
+
+    public void testSomethign() throws IOException {
+        for (int runs = 0; runs < NUMBER_OF_TEST_RUNS; runs++) {
+            TestInstance t = new TestInstance();
+            boolean atRandomPosition = false;
+            for (int i = 0; i < 5; i++) {
```

Comment:
My guess is these repetitions are here so the likelihood for a random shuffle that inserts something in the 1st position that is not the original "field" field is increased? In that case, I think relying on "we just need to do enough repetitions" for a test to pass is trappy and should be avoided.

Reply:
Yup, I need at least on of the iterations to make sure that something is inserted at the 1st position.

I also do not like the test itself as it is rather 'ugly' and testing the test but at the same time it make sense to have it to detect if the behavior is changed ...
I am fine either way ( and removing a test is easy ;) )